### DISCUSSION TITLE
How to insert an image using a relative path

### DISCUSSION THREAD

**Discussion by Equiphract** [freeplane/freeplane#919] [Help]
How can one insert an image into a map by using a relative path to said image instead of the absolute path?
I've been looking around various forums and the help page of FreePlane but I was not able to figure out how to do it.

**Comment by dpolivaev** [comment]
It is a user preference that is applied when you insert an image

Preferences…->Environment->Hyperlink types->Links

**Reply by Equiphract** [reply]
I saw that one and it is already set to relative:
![grafik](https://user-images.githubusercontent.com/31915427/210385136-e0bd588f-0c4f-4baa-bcdb-271883aee3ef.png)

However the images I insert still seem to be linked in an absolute way as when I move the directory containing the map file and the image the link is broken when I open the map up again:
![grafik](https://user-images.githubusercontent.com/31915427/210385796-b3c165c8-7552-483a-994e-2864872131c1.png)

**Reply by dpolivaev** [reply]
How exactly do you insert the images?

**Reply by Equiphract** [reply]
This one in particular I inserted by _right-clicking on the node -> Insert -> Add image -> Add image..._ and then selecting the image via the file browser that opened up.

**Reply by lazycloud314** [reply]
> How exactly do you insert the images?
hi, my freeplane version is 1.10.6 and i tried a bit with relative path.
It seems that Preferences->Environment->Hyperlink types->Links -> Relative works for files that added by Insert -> Link -> Add hyperlink(file), but not for images that added by Insert -> Image -> Add Image

**Reply by flochir** [reply]
Same here, the _Hyperlink types->Links_ setting does not affect image links, those are always of the form
`<hook URI="file:/some/absolute/path/test.png" ... />`
 in the XML code.

The "relative" setting for _Hyperlink types->Links_works on _Insert -> Link -> Add hyperlink (file)_ though, resulting in e.g.
`<node ... LINK="test.pdf"/>`

When I manually edit the .mm files XML to force relative image paths, it opens fine in freeplane, displays the images correctly and also keeps the relative paths when I save it again:

`<hook URI="test.png" ... />`

It would be great if image insert would respect the _Hyperlink types->Links_ setting, or had a equivalent option - _Hyperlink types->Images_?

The absolute paths garble the inserted images when the file is opened from different machines (I want to work with Freeplane using a network drive, accessing it from Linux, Mac and Windows systems).

It might also be a nice additional feature to set the "relative/absolute Links/Image paths" option on the mind map file itself (so an accessing user with "absolute" set in his preferences doesn't produce broken links for the others).

Tested with Freeplane 1.8.0_352 on Linux.

**Comment by ldmpub** [comment]
Hi all,

Other subject on the same topic : **Image and relative link** ...

Scenario : 
1. I rename a Mind Map file (for instance to include a version number: before `1.0` then `1.5` in its filename)
2. I open and add a new image in MM 1.5 => a new "images folder" is automatically created ... resulting in a MindMap with mix / several "images folders" 

Do you know a better way to handle images when changing filename?

Thanks for your tips,

**Reply by AlexandreBonneau** [reply]
---_Maybe I'm missing it, but once I encounter a broken link to an image, I'd like to be able to right click on it, then select 'Edit link properties...' option, so that the user could go select a new (static or relative) path to the image._---

Nevermind; if you right click the broken link, and *not* the image label part, then there is a 'replace...' option that allows just that. Thanks devs ;)
